frontend/src/components/SideBar/GroupIcon.tsx

```diff
@@ -1,21 +1,41 @@
 import React, { ReactElement } from 'react';
+import { connect } from 'react-redux';
+import { State, Tag } from 'common/types/store-types';
+import { NONE_TAG } from 'common/util/tag-util';
 import { Views } from './types';
 import styles from './GroupIcon.module.scss';
+import { getOrderedTags } from '../../store/selectors';
 
-type Props = {
+type OwnProps = {
   classCode: string;
   handleClick: (selectedView: Views, selectedGroup: string | undefined, e?: KeyboardEvent) => void;
   selected: boolean;
 };
 
-const GroupIcon = ({ classCode, handleClick, selected }: Props): ReactElement => (
-  <button
-    type="button"
-    onClick={() => handleClick('group', classCode)}
-    className={styles.GroupIcon + (selected ? ` ${styles.Active}` : '')}
-  >
-    {classCode.charAt(0)}
-  </button>
-);
+type Props = OwnProps & { readonly tags: Tag[] };
 
-export default GroupIcon;
+const GroupIcon = ({ classCode, handleClick, selected, tags }: Props): ReactElement => {
+  const currentClassTag =
+    tags.find(({ classId }) => {
+      const segments = classId !== null ? classId.split(/\s+/) : null;
+      if (segments === null) {
+        return false;
+      }
+      const classIdWithoutNumerical = `${segments[1]} ${segments[2]}`;
+      return classIdWithoutNumerical === classCode;
+    }) ?? NONE_TAG;
```

Comment:
I have seen similar code like this in #590.
Not push blocking for now, but I think it's worth to factor it out to be a custom hook, using the [`useSelector` hook](https://react-redux.js.org/api/hooks#useselector-examples).

Reply:
oh just realized that this diff contains the already merged commit from #590. Merging in latest master or rebasing can make this change disappear.